rust/actyx/swarm/cli/src/main.rs

```diff
@@ -126,19 +124,31 @@ async fn run() -> Result<()> {
     // Poor man's fix for missing ipfs_embed::Event::Connected and
     // ipfs_embed::Event::ConnectionEstablished event from ipfs.swarm_events()
     tokio::spawn(async move {
+        use std::collections::HashSet;
         use tokio::time::{sleep, Duration};
-        let mut connected_peer_ids = std::collections::HashSet::<PeerId>::new();
         let ipfs = ipfs.clone();
-        let listen_addresses = listen_addresses2;
-        while connected_peer_ids.len() < listen_addresses.len() {
-            ipfs.connections().iter().for_each(|connection| {
-                let peer_id = connection.0;
-                let is_connected = ipfs.is_connected(&peer_id);
-                if is_connected && !connected_peer_ids.contains(&peer_id) {
-                    println!("{}", Event::Connected(peer_id));
-                    connected_peer_ids.insert(peer_id);
-                }
+        let mut last_connected = HashSet::<PeerId>::new();
+        loop {
+            let current_connected = ipfs
+                .connections()
+                .iter()
+                .filter(|connection| ipfs.is_connected(&connection.0))
```

Comment:
does this filter ever reject an entry? I’d think that the presence of a connection indicates that we are connected

Reply:
Ah you're right.